### PR TITLE
Adds default order for Daru::DataFrame (issue #130)

### PR DESCRIPTION
**Pull request by athityakumar**
Rspec test added, has been written in the older format to main consistency. This will be easier to change while re-writing all tests to modern specifications file-wise, rather than modernizing specs for individual functions. RDoc documentation has also been added.

Tackles issue #130

### CODE REVIEW

File: lib/daru/dataframe.rb [hunk 1]
```diff
@@ -2278,8 +2289,10 @@ def initialize_from_array source, vectors, index, opts
 
       case source.first
       when Array
+        vectors = (0..source.size-1).to_a if vectors.nil?
```
Comment (zverok) [inline lib/daru/dataframe.rb]: Why not `||=`?

**Comment by athityakumar** [comment]
@zverok @v0dro - After closer inspection of the logs of the last successful build and current failing build of Travis CI, I observe that `bundle install` used to install `rubocop v0.47.1` (build passing) and is now installing `rubocop v0.48.0` (build failing). Have a look at the respective [releases](https://github.com/bbatsov/rubocop/releases) of rubocop gem.

We can specify `rubocop (~>= 0.47.1)` in the `Gemfile.lock` and remove the `Gemfile.lock` from `.gitignore`. Do you think this would fix the build error, or could it be something else?

**Comment by zverok** [comment]
> We can specify rubocop (~>= 0.47.1) in the Gemfile.lock and remove the Gemfile.lock from .gitignore. Do you think this would fix the build error, or could it be something else?

Instead of that, we can fix **current** rubocop version offences on each PR. They are reasonable, and being a small project we can't expect dedicated volunteers to "update rubocop" in separate PR. So whoever met new offences first is that unfortunate person who fixes them :)

**Comment by zverok** [comment]
(but you can exclude Rakefile and Gemfile from rubocop-checked files, of course)

**Comment by athityakumar** [comment]
@zverok - Alright, I'm taking one for the team. :)

Rubocop offenses are fixed, and build is finally successful. It does feel good to see the `Build passing` and green tick again. :tada:

File: lib/daru/dataframe.rb [hunk 2]
```diff
@@ -244,6 +244,17 @@ def from_vector_rows source, opts
     #   #  b          7          2
     #   #  c          8          3
     #   #  d          9          4
+    #
+    #   df = Daru::DataFrame.new([[1,2,3,4],[6,7,8,9]], name: :bat_man)
+    #
+    #   # =>
+    #   # #<Daru::DataFrame: bat_man (4x2)>
+    #   #             0          1
```
Comment (v0dro) [inline lib/daru/dataframe.rb]: Given your example, shouldn't `[6,7,8,9]` be the 1st vector?

Comment (athityakumar) [inline lib/daru/dataframe.rb]: Thanks for pointing this out - fixed! :smile:

Comment (v0dro) [inline lib/daru/dataframe.rb]: Didn't you use actual executed code for the example? Its better if you do that.

Comment (athityakumar) [inline lib/daru/dataframe.rb]: I hadn't used the actual executed code in the initial commit (I had copy-pasted the previous `rdoc` example - my bad). I've added the output I get in the `irb` console while initializing the DataFrame given in the `rdoc` example, in the subsequent commit.

**Pull request merged**